app/Controllers/UsersController.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controllers;
+
+use App\View;
+use App\Models\User;
+
+
+class UsersController
+{
+    CONST REQUEST_METHOD = 'POST';
+    protected $userModel;
+
+    public function __construct()
+    {
+        $this->userModel = new User();
+    }
+
+
+    /**
+     * function to register a new user
+     * @return View|void
+     */
+    public function register()
+    {
+        if ($_SERVER['REQUEST_METHOD'] == self::REQUEST_METHOD) {
+            $_POST = filter_input_array(INPUT_POST, FILTER_SANITIZE_STRING);
+            $data = [
+                'name' => trim($_POST['name']),
+                'email' => trim($_POST['email']),
+                'password' => trim($_POST['password']),
+                'confirm_password' => trim($_POST['confirm_password']),
+                'name_err' => '',
+                'email_err' => '',
+                'password_err' => '',
+                'confirm_password_err' => ''
+            ];
+
```

Comment:
This validation below could possibly be used in another function like validateUser($data)